### ISSUE TITLE
Zooming wireframe shows extra lines, regression Makie 0.19 -> 0.20

### ISSUE THREAD

**Issue by jaakkor2** [MakieOrg/Makie.jl#3544]
- [x] are you running newest version: yes, GLMakie v0.9.5, Makie v0.20.4
- [x] can you reproduce the bug with a fresh environment ? yes
- [x] What platform + GPU are you on?  Windows 10, Intel integrated i7-1370P

```
# adapted from https://docs.makie.org/stable/reference/plots/wireframe/

using GLMakie

x, y = collect(-8:0.5:8), collect(-8:0.5:8)
z = [sinc(√(X^2 + Y^2) / π) for X ∈ x, Y ∈ y]
wf = wireframe(x, y, z, color=:black)
wf.axis.show_axis[] = false
zoom!(wf.axis.scene, 0.1)
save("Makie0p20.png", wf.figure, update=false)
```
shows
![Makie0p20](https://github.com/MakieOrg/Makie.jl/assets/7318249/58a4e606-b0f0-4d37-b9a2-e23413e13554)

The wireframe looks fine when zooming less.

**Comment by ffreyer** [comment]
There is also an issue with axis lines shifting/inverting when zooming in. Maybe this is another version of that problem?

**Comment by ffreyer** [comment]
Fixed by #3651 via #3650 in 0.21
